juju/juju
# Create interactive add-cloud command

This PR adds an interactive way to add a cloud via answering prompts
rather than having to specify a premade yaml file.

It also includes changes to the interact package to support 
standardized ways of querying the user for information, which
can be reused for other interactive commands.

QA:
run `juju add-cloud`
Should show options to create clouds of types maas, manual, openstack, and vsphere.   
Should prompt for all required information to add the cloud, which should be added to your clouds.yaml.
You should then be able to use that cloud definition with juju bootstrap.
